fitbenchmarking/results_processing/base_table.py

```diff
@@ -380,15 +403,26 @@ def to_html(self):
         table.columns = columns
 
         # Format the row labels
-        index = []
-        for b, i in zip(self.best_results.values(), table.index):
+        index1 = []
+        for b, i in zip(self.best_results.values(), self.best_results.keys()):
+            b = next(iter(b.values()))
+            rel_path = os.path.relpath(
+                path=b.problem_summary_page_link,
+                start=self.group_dir)
+            index1.append(f'<a class="problem_header" '
+                          f'href="{rel_path}">{i}</a>')
+
+        index2 = []
+        for b, i in zip(self.best_results.values(), self.problem_sizes):
             b = next(iter(b.values()))
             rel_path = os.path.relpath(
                 path=b.problem_summary_page_link,
                 start=self.group_dir)
-            index.append(f'<a class="problem_header" href="{rel_path}">{i}</a>'
-                         )
-        table.index = index
+            index2.append(f'<a class="problem_header_lev1" '
+                          f'href="{rel_path}">{i}</a>')
```

Comment:
I think both `index1` and `index2` can be updated in the same loop. If we make `index` have entries which are tuples then there's no need to zip. `index.append((f'<a class="problem_header"...', ...))`